File: routes/admin/userlist.js

var res = require('./lib/resp.js');
var cates = require("./models/cates.js");
var users = require("./models/users.js");

if (!session.get("isAdmin")) {
  return res.redirect('/routes/admin/login.js');
}

// var allData = cates.find();
// var data = {
//   tree: cates.getTree(allData),
// };
// res.render('./views/admin/catelist.ejs', data);

var us = users.find({}, {});
res.render('./views/admin/userlist.ejs', { users: us });

File: routes/admin/userlist.go
package admin

import (
	"github.com/zengming00/go-testShop/framework"
)

func UserList(ctx *framework.HandlerContext) {
	var isAdmin = false
	if v, ok := ctx.GetSessionVal("isAdmin"); ok {
		isAdmin = v.(bool)
	}
	if !isAdmin {
		ctx.Redirect("/routes/admin/login.go")
	}

	var us = ctx.Users.Find(nil, nil)
	var data = map[string]interface{}{
		"users": us,
	}
	ctx.Render("./views/admin/userlist.html", data, nil)
}
